miniProxy.php

```diff
@@ -153,10 +153,20 @@ function($matches) use ($baseURL) {
     $css);
 }
 
+// Parse/extract URL
 $url = substr($_SERVER["REQUEST_URI"], strlen($_SERVER["SCRIPT_NAME"]) + 1);
-if (empty($url)) die("<html><head><title>miniProxy</title></head><body><h1>Welcome to miniProxy!</h1>miniProxy can be directly invoked like this: <a href=\"" . PROXY_PREFIX . "http://google.com/\">" . PROXY_PREFIX . "http://google.com/</a><br /><br />Or, you can simply enter a URL below:<br /><br /><form onsubmit=\"window.location.href='" . PROXY_PREFIX . "' + document.getElementById('site').value; return false;\"><input id=\"site\" type=\"text\" size=\"50\" /><input type=\"submit\" value=\"Proxy It!\" /></form></body></html>");
-if (strpos($url, "//") === 0) $url = "http:" . $url; //Assume that any supplied URLs starting with // are HTTP URLs.
-if (!preg_match("@^.*://@", $url)) $url = "http://" . $url; //Assume that any supplied URLs without a scheme are HTTP URLs.
+if (empty($url)) {
+    die("<html><head><title>miniProxy</title></head><body><h1>Welcome to miniProxy!</h1>miniProxy can be directly invoked like this: <a href=\"" . PROXY_PREFIX . "http://google.com/\">" . PROXY_PREFIX . "http://google.com/</a><br /><br />Or, you can simply enter a URL below:<br /><br /><form onsubmit=\"window.location.href='" . PROXY_PREFIX . "' + document.getElementById('site').value; return false;\"><input id=\"site\" type=\"text\" size=\"50\" /><input type=\"submit\" value=\"Proxy It!\" /></form></body></html>");
+} else if (strpos($url, "//") === 0) {
+    $url = "http:" . $url; //Assume that any supplied URLs starting with // are HTTP URLs.
+} else if (strpos($url, ':/') !== strpos($url, '://')) {
```

Comment:
Could this condition be simplified to `(strpos($url, '://') === -1)`?

Either way, this would seem to attempt to check the scheme if one exists, rather than the query string.

If there isn't a scheme, wouldn't it be possible for this to cause issues in the unlikely case that the query string contains occurrences of both `:/` and `://` for some reason?